test/integration/node-specific/resource_tracking_script_builder.ts

```diff
@@ -106,3 +132,70 @@ export async function runScript(
     heap
   };
 }
+
+/**
+ * A helper for running arbitrary MongoDB Driver scripting code in a resource information collecting script.
+ * This script uses info from node:process to collect resource information.
+ *
+ * **The provided function is run in an isolated Node.js process**
+ *
+ * A user of this function will likely need to familiarize themselves with the surrounding scripting, but briefly:
+ * - Many MongoClient operations (construction, connection, commands) can result in resources that keep the JS event loop running.
+ *   - Timers
+ *   - Active Sockets
+ *   - File Read Hangs
+ *
+ * - This function performs assertions that at the end of the provided function, the js event loop has been exhausted
+ *
+ * @param name - the name of the script, this defines the name of the file, it will be cleaned up if the function returns successfully
+ * @param config - `this.configuration` from your mocha config
+ * @param func - your javascript function, you can write it inline! this will stringify the function, use the references on the `this` context to get typechecking
+ * @param options - settings for the script
+ * @throws Error - if the process exits with failure or if the process' resources are not cleaned up by the provided function.
+ */
+export async function runScriptAndGetProcessInfo(
+  name: string,
+  config: TestConfiguration,
+  func: ProcessResourceTestFunction
+) {
+  const scriptName = `${name}.cjs`;
+  const scriptContent = await testScriptFactory(
+    name,
+    config.url(),
+    REPORT_RESOURCE_SCRIPT_PATH,
+    func
+  );
+  await writeFile(scriptName, scriptContent, { encoding: 'utf8' });
+  const logFile = name + '.logs.txt';
+
+  const script = spawn(process.argv[0], [scriptName], { stdio: ['ignore', 'ignore', 'ignore'] });
+
+  const willClose = once(script, 'close');
+
+  // make sure the process ended
+  const [exitCode] = await willClose;
+
+  // format messages from child process as an object
+  const messages = (await readFile(logFile, 'utf-8'))
+    .trim()
+    .split('\n')
+    .map(line => JSON.parse(line))
+    .reduce((acc, curr) => ({ ...acc, ...curr }), {});
```

Comment:
We discussed that this "reduce" was squashing the "errorCode" field (which is now gone), we still reuse the "error" field in two places, because we exit we shouldn't squash it, but should we use the array so we don't have the potential to make that mistake? 

Don't feel strongly here, let me know your thoughts.

Reply:
I'd rather just keep the object here since I find it easier to parse. If we come across an array case being needed it'll be a 2 line change we can make then. What do you think?